### ISSUE TITLE
Prepare for CNCF Incubation application

### ISSUE THREAD

**Issue by viccuad** [kubewarden/community#11]
### Acceptance criteria

We need to fill the following [template for incubation application](https://github.com/cncf/toc/blob/main/.github/ISSUE_TEMPLATE/template-incubation-application.md) (2024-07-04 v1.5 as time of writing).
This issue contains a copy of that template so we can iterate and piecemeal the tasks.

To work on this epic, pick one of the required tasks, convert it into an issue and move it to the relevant repo (normally kubewarden/community).
Once the required tasks are done, we will format this issue into a submission.

Suggestion for scheduling work: "complete X number of tasks from the list", as there are still some low hanging fruits.

---

# Kubewarden Incubation Application
v1.5

Project Repo(s): https://github.com/kubewarden/community
Project Site:  https://kubewarden.io
Sub-Projects: [$LIST](https://github.com/kubewarden/community?tab=readme-ov-file#repositories)
Communication: [#kubewarden](https://kubernetes.slack.com/archives/C01T3GTC3L7) on Kubernetes Slack, [#kubewarden-dev](https://kubernetes.slack.com/archives/C03L52JRAFM) on Kubernetes Slack, cncf-kubewarden-maintainers as mail username at lists.cncf.io.

Project points of contacts: https://github.com/orgs/kubewarden/teams/maintainers

### Adoption Assertion

_The project has been adopted by the following organizations in a testing and integration or production capacity:_
(list adopters in [ADOPTERS.md](https://github.com/kubewarden/community/blob/main/ADOPTERS.md))

### Suggested

N/A

### Required

- [ ] **Give a presentation and engage with the domain specific TAG(s) to increase awareness**
  - This was completed and occurred on DD-MMM-YYYY, and can be discovered at $LINK.

 

- [ ] **TAG provides insight/recommendation of the project in the context of the landscape**

 

- [x] **All project metadata and resources are [vendor-neutral](https://contribute.cncf.io/maintainers/community/vendor-neutrality/).**

 

- [x] **Review and acknowledgement of expectations for [Sandbox](https://sandbox.cncf.io) projects and requirements for moving forward through the CNCF Maturity levels.**		
- Met during Project's [application on 27-04-2024](https://github.com/cncf/toc/issues/858).

 

- [ ] **[Due Diligence Review.](https://github.com/cncf/toc/blob/f41d808146626b7e76bd94d69ffb4ed976f17de3/process/due-diligence-guidelines.md)**

Completion of this due diligence document, resolution of concerns raised, and presented for public comment satisifies the Due Diligence Review criteria.

- [x] **Additional documentation as appropriate for project type, e.g.: installation documentation, end user documentation, reference implementation and/or code samples.**
  https://kubewarden.io
  https://docs.kubewarden.io
 

## Governance and Maintainers

Note: this section may be augmented by the completion of a Governance Review from TAG Contributor Strategy.

### Suggested

- [x] **Clear and discoverable project governance documentation.**
  https://github.com/kubewarden/community/blob/main/GOVERNANCE.md

 

- [ ] **Governance has continuously been iterated upon by the project as a result of their experience applying it, with the governance history demonstrating evolution of maturity alongside the project's maturity evolution.**

 

- [x] **Governance is up to date with actual project activities, including any meetings, elections, leadership, or approval processes.**

 

- [ ] **Governance clearly documents [vendor-neutrality](https://contribute.cncf.io/maintainers/community/vendor-neutrality/) of project direction.**

 

- [ ] **Document how the project makes decisions on leadership, contribution acceptance, requests to the CNCF, and changes to governance or project goals.**

 

- [ ] **Document how role, function-based members, or sub-teams are assigned, onboarded, and removed for specific teams (example: Security Response Committee).**

 

- [ ] **Document a complete maintainer lifecycle process (including roles, onboarding, offboarding, and emeritus status).**

 

- [ ] **Demonstrate usage of the maintainer lifecycle with outcomes, either through the addition or replacement of maintainers as project events have required.**

 

- [ ] **If the project has subprojects: subproject leadership, contribution, maturity status documented, including add/remove process.**

 

### Required

- [ ] **Document complete list of current maintainers, including names, contact information, domain of responsibility, and affiliation.**

 

- [ ] **A number of active maintainers which is appropriate to the size and scope of the project.**

 

- [ ] **Code and Doc ownership in Github and elsewhere matches documented governance roles.**

 

- [x] **Document agreement that project will adopt CNCF Code of Conduct.**
  https://github.com/kubewarden/community/blob/main/CODE_OF_CONDUCT.md

 

- [ ] **CNCF Code of Conduct is cross-linked from other governance documents.**

 

- [x] **All subprojects, if any, are listed.**
  https://github.com/kubewarden/community/blob/main/README.md#repositories

 

## Contributors and Community

Note: this section may be augmented by the completion of a Governance Review from TAG Contributor Strategy.
- [ ] (Optional) Submit a Governance Review from TAG Contributor Strategy.

### Suggested

- [ ] **Contributor ladder with multiple roles for contributors.**

 

### Required

- [x] **Clearly defined and discoverable process to submit issues or changes.**
  We use GitHub for submitting issues and changes via PRs. Contributors can get in contact with us at [#kubewarden-dev](https://kubernetes.slack.com/archives/C03L52JRAFM) on the Kubernetes Slack server, as well as [cncf-kubewarden-maintainers@lists.cncf.io](mailto:cncf-kubewarden-maintainers@lists.cncf.io).

 

- [x] **Project must have, and document, at least one public communications channel for users and/or contributors.**
  We use [#kubewarden-dev](https://kubernetes.slack.com/archives/C03L52JRAFM), [#kubewarden](https://kubernetes.slack.com/archives/C01T3GTC3L7) on the Kubernetes Slack server, as well as [cncf-kubewarden-maintainers@lists.cncf.io](mailto:cncf-kubewarden-maintainers@lists.cncf.io). These are linked in https://kubewarden.io.
 

- [x] **List and document all project communication channels, including subprojects (mail list/slack/etc.).  List any non-public communications channels and what their special purpose is.**
  - [#kubewarden](https://kubernetes.slack.com/archives/C01T3GTC3L7) on the Kubernetes Slack server. Main public channel, for user questions, announcements and general talk.
  - [#kubewarden-dev](https://kubernetes.slack.com/archives/C03L52JRAFM) on the Kubernetes Slack server. Development channel for development questions, PR reviews, etc.
  - [cncf-kubewarden-maintainers@lists.cncf.io](mailto:cncf-kubewarden-maintainers@lists.cncf.io). Mailing list. For private Security disclosures.
 

- [ ] **Up-to-date public meeting schedulers and/or integration with CNCF calendar.**
  - [ ] TODO add Kubewarden monthly community meetings to https://www.cncf.io/calendar/.

 

- [x] **Documentation of how to contribute, with increasing detail as the project matures.**
  https://github.com/kubewarden/community/blob/main/CONTRIBUTING.md

 

- [ ] **Demonstrate contributor activity and recruitment.**

 

### Suggested

- [ ] **Roadmap change process is documented.**

 

- [x] **History of regular, quality releases.**
  We release somewhat monthly, with release candidates. For more information on each release, besides GitHub Releases and their changelogs, see https://www.kubewarden.io/blog/.

 

### Required 

- [ ] **Document project goals and objectives that illustrate the project’s differentiation in the Cloud Native landscape as well as outlines how this project fulfills an outstanding need and/or solves a problem differently.**

 

- [ ] **Document what the project does, and why it does it - including viable cloud native use cases.**

 

- [ ] **Document and maintain a public roadmap or other forward looking planning document or tracking mechanism.**

 

- [ ] **Document overview of project architecture and software design that demonstrates viable cloud native use cases, as part of the project's documentation.**

 

- [ ] **Document the project's release process.**

 

## Security

Note: this section may be augmented by a joint-assessment performed by TAG Security.
- [ ] (optional) Submit for an assessment by TAG Security

### Suggested

N/A

### Required

- [x] **Clearly defined and discoverable process to report security issues.**
  https://docs.kubewarden.io/disclosure. Also, https://github.com/kubewarden/community/blob/main/SECURITY.md.

 

- [ ] **Enforcing Access Control Rules to secure the code base against attacks (Example: two factor authentication enforcement, and/or use of ACL tools.)**

 

- [ ] **Document assignment of security response roles and how reports are handled.**

 

- [ ] **Document Security Self-Assessment.**

 

- [ ] **Achieve the Open Source Security Foundation (OpenSSF) Best Practices passing badge.**
  All the MUST and SHOULD must be achieved. See criteria in https://www.bestpractices.dev/en/criteria/0.
  Current progress in https://www.bestpractices.dev/en/users/23101. 
  - https://github.com/kubewarden/policy-server/issues/810
  - https://github.com/kubewarden/kwctl/issues/851

 

### Suggested

N/A

### Required

- [x] **Publicly documented list of adopters, which may indicate their adoption level (dev/trialing, prod, etc.)**
  https://github.com/kubewarden/community/blob/main/ADOPTERS.md
 

- [x] **Used in appropriate capacity by at least 3 independent + indirect/direct adopters, (these are not required to be in the publicly documented list of adopters)**
   
  The project provided the TOC with a list of adopters for verification of use of the project at the level expected, i.e. production use for graduation, dev/test for incubation: https://github.com/kubewarden/community/blob/main/ADOPTERS.md

- [ ] **TOC verification of adopters.**

 

Refer to the Adoption portion of this document.

- [ ] **Clearly documented integrations and/or compatibility with other CNCF projects as well as non-CNCF projects.**
 Docs page listing the dependencies used: https://docs.kubewarden.io/reference/dependency-matrix

**Comment by flavio** [comment]
@orlinvasilev: could you help with that when you've some spare time, please? 🙏

**Comment by fabriziosestito** [comment]
<details>
<summary>Kubewarden Incubation Application</summary>

---
name: Project Incubation Application
about: This template provides the project with a framework to inform the TOC of their conformance to the Incubation Level Criteria.
title: "[Incubation] $PROJECT Incubation Application"
labels: incubation
---

# Review Project Moving Level Evaluation
[ ] I have reviewed the TOC's [moving level readiness triage guide](https://github.com/cncf/toc/blob/main/operations/dd-toc-guide.md#initial-triageevaluation-prior-to-assignment), ensured the criteria for my project are met before opening this issue, and understand that unmet criteria will result in the project's application being closed.

# Kubewarden Incubation Application
v1.6
This template provides the project with a framework to inform the TOC of their conformance to the Incubation Level Criteria. 

Project Repo(s): https://github.com/kubewarden/community
Project Site:  https://kubewarden.io/
Sub-Projects: https://github.com/kubewarden/community?tab=readme-ov-file#repositories
Communication: [#kubewarden](https://kubernetes.slack.com/archives/C01T3GTC3L7) on Kubernetes Slack, [#kubewarden-dev](https://kubernetes.slack.com/archives/C03L52JRAFM) on Kubernetes Slack, cncf-kubewarden-maintainers as mail username at lists.cncf.io.

Project points of contacts: https://github.com/orgs/kubewarden/teams/maintainers or cncf-kubewarden-maintainers as mail username at lists.cncf.io.

- [ ] (Post Incubation only) [Book a meeting with CNCF staff](http://project-meetings.cncf.io) to understand project benefits and event resources. 

## Incubation Criteria Summary for Kubewarden

### Application Level Assertion

- [x] This project is currently Sandbox, accepted on 2022-06-14, and applying to Incubation.
- [x] This project is applying to join the CNCF at the Incubation level.

### Adoption Assertion

_The project has been adopted by the following organizations in a testing and integration or production capacity:_

See [ADOPTERS.md](https://github.com/kubewarden/community/blob/main/ADOPTERS.md)

## Application Process Principles

### Suggested

N/A

### Required

- [x] **Engage with the domain specific TAG(s) to increase awareness through a presentation or completing a General Technical Review.**
  - This was completed and occurred on 09-Jun-2022, and can be discovered at https://github.com/cncf/tag-security/issues/899.
    slide deck: https://drive.google.com/file/d/1JeC0GsDHP0X5dnTjhZTAL_Ruc1y12PJt/view?usp=sharing
    recording: https://www.youtube.com/watch?v=K-8Nepk7Zyc
    General Technical Review: https://github.com/kubewarden/community/blob/main/general-technical-questions.md

 

- [ ] **TAG provides insight/recommendation of the project in the context of the landscape**

 

- [x] **All project metadata and resources are [vendor-neutral](https://contribute.cncf.io/maintainers/community/vendor-neutrality/).**

 

- [x] **Review and acknowledgement of expectations for [Sandbox](https://sandbox.cncf.io) projects and requirements for moving forward through the CNCF Maturity levels.**		
- Met during Project's application on 14-Jul-2022

 
https://github.com/cncf/sandbox/issues/213

https://lists.cncf.io/g/cncf-toc/topic/results_from_sandbox/91754161?p=,,,20,0,0,0::recentpostdate/sticky,,,20,2,0,91754161,previd%3D1655255497954310253,nextid%3D1653320044101394801&previd=1655255497954310253&nextid=1653320044101394801

- [ ] **Due Diligence Review.**

Completion of this due diligence document, resolution of concerns raised, and presented for public comment satisfies the Due Diligence Review criteria.

- [x] **Additional documentation as appropriate for project type, e.g.: installation documentation, end user documentation, reference implementation and/or code samples.**

 

https://docs.kubewarden.io/
https://github.com/kubewarden/community

## Governance and Maintainers

Note: this section may be augmented by the completion of a Governance Review from TAG Contributor Strategy.

### Suggested

- [x] **Clear and discoverable project governance documentation.**

 
https://github.com/kubewarden/community/blob/main/GOVERNANCE.md

- [x] **Governance has continuously been iterated upon by the project as a result of their experience applying it, with the governance history demonstrating evolution of maturity alongside the project's maturity evolution.**

 

- [x] **Governance is up to date with actual project activities, including any meetings, elections, leadership, or approval processes.**

 

- [x] **Governance clearly documents [vendor-neutrality](https://contribute.cncf.io/maintainers/community/vendor-neutrality/) of project direction.**

 
https://docs.kubewarden.io/#vendor-neutrality

- [x] **Document how the project makes decisions on leadership, contribution acceptance, requests to the CNCF, and changes to governance or project goals.**

 

https://github.com/kubewarden/community/blob/main/GOVERNANCE.md

- [ ] **Document how role, function-based members, or sub-teams are assigned, onboarded, and removed for specific teams (example: Security Response Committee).**

 

- [x] **Document a complete maintainer lifecycle process (including roles, onboarding, offboarding, and emeritus status).**

 

In the Kubewarden team, we have only two roles that are applied to the whole project. These role definitions are defined in the [GOVERNANCE.md](https://github.com/kubewarden/community/blob/main/GOVERNANCE.md) file

- [x] **Demonstrate usage of the maintainer lifecycle with outcomes, either through the addition or replacement of maintainers as project events have required.**

 

https://github.com/kubewarden/community/blob/main/GOVERNANCE.md

- [x] **If the project has subprojects: subproject leadership, contribution, maturity status documented, including add/remove process.**

 

### Required

- [x] **Document complete list of current maintainers, including names, contact information, domain of responsibility, and affiliation.**

 

https://github.com/kubewarden/community/blob/main/MAINTAINERS.md

- [x] **A number of active maintainers which is appropriate to the size and scope of the project.**

 

https://github.com/kubewarden/community/blob/main/MAINTAINERS.md

- [x] **Code and Doc ownership in Github and elsewhere matches documented governance roles.**

 

We use the CODEOWNERS file in all of the repositories. Some examples:
https://github.com/kubewarden/kubewarden-controller/blob/main/CODEOWNERS
https://github.com/kubewarden/docs/blob/main/CODEOWNERS

- [x] **Document adoption of the CNCF Code of Conduct**

 

https://github.com/kubewarden/community/blob/main/CODE_OF_CONDUCT.md

- [x] **CNCF Code of Conduct is cross-linked from other governance documents.**

 

https://github.com/kubewarden/community/blob/main/CODE_OF_CONDUCT.md

- [x] **All subprojects, if any, are listed.**

 

https://github.com/kubewarden/community?tab=readme-ov-file#repositories

## Contributors and Community

Note: this section may be augmented by the completion of a Governance Review from TAG Contributor Strategy.

### Suggested

- [ ] **Contributor ladder with multiple roles for contributors.**

 

### Required

- [x] **Clearly defined and discoverable process to submit issues or changes.**

 

We use GitHub to submit issues and changes via PRs. Contributors can get in contact with us at [#kubewarden-dev](https://kubernetes.slack.com/archives/C03L52JRAFM) on the Kubernetes Slack server, as well as [cncf-kubewarden-maintainers@lists.cncf.io](mailto:cncf-kubewarden-maintainers@lists.cncf.io). We also have the [CONTRIBUTING.md](https://github.com/kubewarden/community/blob/main/CONTRIBUTING.md) file

- [x] **Project must have, and document, at least one public communications channel for users and/or contributors.**

 

We use [#kubewarden-dev](https://kubernetes.slack.com/archives/C03L52JRAFM), [#kubewarden](https://kubernetes.slack.com/archives/C01T3GTC3L7) on the Kubernetes Slack server, as well as [cncf-kubewarden-maintainers@lists.cncf.io](mailto:cncf-kubewarden-maintainers@lists.cncf.io). These are linked in https://kubewarden.io/. We also describe how to contribute to the project in the [CONTRIBUTING.md](https://github.com/kubewarden/community/blob/main/CONTRIBUTING.md#feedback) file

- [x] **List and document all project communication channels, including subprojects (mail list/slack/etc.).  List any non-public communications channels and what their special purpose is.**

 

https://github.com/kubewarden/community/tree/main?tab=readme-ov-file#community

- [x] **Up-to-date public meeting schedulers and/or integration with CNCF calendar.**

We have our community calls listed in the [CNCF calendar](https://www.cncf.io/calendar/) and in our [community repository](https://github.com/kubewarden/community/tree/main?tab=readme-ov-file#community-meeting)

- [x] **Documentation of how to contribute, with increasing detail as the project matures.**

https://github.com/kubewarden/community/blob/main/CONTRIBUTING.md
https://github.com/kubewarden/kubewarden-controller/blob/main/CONTRIBUTING.md

- [x] **Demonstrate contributor activity and recruitment.**

https://kubewarden.devstats.cncf.io/
https://kubewarden.devstats.cncf.io/d/74/contributions-chart?orgId=1

## Engineering Principles

### Suggested

- [x] **Roadmap change process is documented.**

 

See https://github.com/kubewarden/community/blob/main/ROADMAP.md

- [x] **History of regular, quality releases.**

We follow a monthly release cadence. The releases, including the release candidates, can be found in the GitHub repositories.
We also write a [blog post](https://www.kubewarden.io/blog/) for each release.
https://github.com/kubewarden/kubewarden-controller/releases
https://github.com/kubewarden/policy-server/releases
https://github.com/kubewarden/kwctl/releases
https://github.com/kubewarden/audit-scanner/releases

### Required 

- [x] **Document project goals and objectives that illustrate the project’s differentiation in the Cloud Native landscape as well as outlines how this project fulfills an outstanding need and/or solves a problem differently. This can also be satisfied by completing a General Technical Review.**

 

 General Technical Review: https://github.com/kubewarden/community/blob/main/general-technical-questions.md

- [x] **Document what the project does, and why it does it - including viable cloud native use cases. This can also be satisfied by completing a General Technical Review.**

 

 General Technical Review: https://github.com/kubewarden/community/blob/main/general-technical-questions.md

- [x] **Document and maintain a public roadmap or other forward looking planning document or tracking mechanism.**

We use Github milestones to define the roadmap for future releases. This can be found in the project board at the "Milestone" tab
https://github.com/orgs/kubewarden/projects/6/views/1
The roadmap strategy is documented here: https://github.com/kubewarden/community/blob/main/ROADMAP.md

- [x] **Document overview of project architecture and software design that demonstrates viable cloud native use cases, as part of the project's documentation. This can also be satisfied by completing a General Technical Review and capturing the output in the project's documentation.**

 

 General Technical Review: https://github.com/kubewarden/community/blob/main/general-technical-questions.md

https://docs.kubewarden.io/explanations/architecture

- [x] **Document the project's release process.**

https://github.com/kubewarden/helm-charts/blob/main/CONTRIBUTING.md
https://github.com/kubewarden/kubewarden-controller/blob/main/CONTRIBUTING.md#kubewarden-release-template

## Security

Note: this section may be augmented by a joint-assessment performed by TAG Security.

### Suggested

N/A

### Required

- [x] **Clearly defined and discoverable process to report security issues.**

 

https://github.com/kubewarden/community/blob/main/SECURITY.md
https://docs.kubewarden.io/disclosure

- [x] **Enforcing Access Control Rules to secure the code base against attacks (Example: two factor authentication enforcement, and/or use of ACL tools.)**

 

As a Github hosted project, we rely on the Github authentication mechanisms. Most of the maintainers use two factor authentication and sign commits and tags with GPG keys

- [x] **Document assignment of security response roles and how reports are handled.**

This is described in the [SECURITY.md](https://github.com/kubewarden/community/blob/main/SECURITY.md) file.

- [x] **Document [Security Self-Assessment](https://tag-security.cncf.io/community/assessments/guide/self-assessment/).**

 

https://github.com/kubewarden/community/blob/main/security-self-assesment.md

- [x] **Achieve the Open Source Security Foundation (OpenSSF) Best Practices passing badge.**

 

https://www.bestpractices.dev/en/users/23101


## Ecosystem

### Suggested

N/A

### Required

- [x] **Publicly documented list of adopters, which may indicate their adoption level (dev/trialing, prod, etc.)**

 

https://github.com/kubewarden/community/blob/main/ADOPTERS.md

- [x] **Used in appropriate capacity by at least 3 independent + indirect/direct adopters, (these are not required to be in the publicly documented list of adopters)**

 

The project provided the TOC with a list of adopters for verification of use of the project at the level expected, i.e. production use for graduation, dev/test for incubation: https://github.com/kubewarden/community/blob/main/ADOPTERS.md


- [ ] **TOC verification of adopters.**

 

Refer to the Adoption portion of this document.

- [x] **Clearly documented integrations and/or compatibility with other CNCF projects as well as non-CNCF projects.**

Docs page listing the dependencies used: https://docs.kubewarden.io/reference/dependency-matrix

## Additional Information



</details>

**Comment by jvanz** [comment]
Folks, I've review the incubation application template and TOC documentation. Most of the items from previous Fabrizio comment are still valid. But the form has new items and I think we can try to increase our chance of moving forward to the incubation level if we are proactive. 

The TOC has has some [requirements](https://github.com/cncf/toc/blob/main/process/README.md#applications-to-move-levels-are-done-by-submitting-an-incubation-or-graduation-application-issue-on-the-toc-repo) that we must have and, besides the filled application form,  they are:

- **Project must complete the [Adopter Interview Form with 5-7 adopters willing to be interviewed](https://docs.google.com/forms/d/1n1oLC6IKj5-7S_xeEjIdEjbtS9SWniuAo7IIOyLFuK8/)**: we are aware of that. But I was not sure if we remember that the number is 5-7 and not 3. Note that the 5-7 adopters should be **willing** to be interviewed. Not only listed in the adopters list. We already discussed that some time. Just sharing here again to keep documented keep everybody in the same page
- **Project must provide a self-assessed Technical Review with their application.**: we already have [this](https://github.com/kubewarden/community/blob/main/general-technical-questions.md).  As this document address many item from the application form, I would suggest to review it and ensure that our document is following the latest version from TOC [template](https://github.com/cncf/toc/blob/main/toc_subprojects/project-reviews-subproject/general-technical-questions.md). Also, ensure that the SBOMScanner is cover in the document as well. 
- **Project must provide a self-assessed Governance Review with their application.**: we do not have this. The template is [here](https://github.com/cncf/toc/blob/main/toc_subprojects/project-reviews-subproject/governance-review-template.md)

About the self-assessed reviews from the TOC [documentation](https://github.com/cncf/toc/blob/main/operations/dd-toc-guide.md#initial-triageevaluation-prior-to-assignment):

> Projects should complete Technical and Governance reviews themselves and store them in a .project folder or GitHub repository in their own organization. CNCF associates will review and upload vetted snapshots of the project's copies to [github.com/cncf/toc/projects](https://github.com/cncf/toc/blob/main/projects), but projects should not be blocked by this process, as it depends on availability of community members in CNCF TAGs which cannot be guaranteed.

> [!NOTE]
> I'm suggesting the review of the current self-assessed document just ensure that they have not get an update as the application form

Besides that I think we can work in other tasks to improve our application feedback:

- Complete a joint security assessment with TAG Security and Compliance. This is a suggestion from the security section of the application form. (optional as stated in the application form)
- Update the SECURITY.md mentioning explicit who and the roles in the process of handling the security incident. This is little improvement to cover a new item from the security section of the application form
- Review self security assessment. As General technical review, we already [did it](https://github.com/kubewarden/community/blob/main/security-self-assesment.md) some time ago.  Just review it to ensure that we are following the latest version from TOC and it should also cover the SBOMScanner component.
- Add SBOMScanner in the [OpenSSF best practices](https://www.bestpractices.dev/en/users/23101)
- While I was reviewing the [TOC board](https://github.com/orgs/cncf/projects/27/views/9) I noticed that card from k3s and a DD review point the [single vendor maintainer dependency](https://github.com/cncf/toc/issues/1957#issuecomment-4872412043). I think we will receive something similar in our application and we could try to mitigate this upfront. I'm not sure if the [suggestion](https://github.com/cncf/toc/issues/1957#issuecomment-4872426815) from DD review is doable in our case. But I believe we should think about it. 

In the next section I'll add the new template with the information copied from Fabrizio [comment](https://github.com/kubewarden/community/issues/11#issuecomment-2823063114) where applicable and with up to date link. As you can see almost all the required and suggested items are already addressed. But the governance review is still missing, as mentioned above:

# Incubation application draft

<details>
<summary> click here to see Incubation application draft</summary>

# Review Project Moving Level Evaluation
- [ ] I have reviewed the TOC's [moving level readiness triage guide](https://github.com/cncf/toc/blob/main/operations/dd-toc-guide.md#initial-triageevaluation-prior-to-assignment), ensured the criteria for my project are met before opening this issue, and understand that unmet criteria will result in the project's application being closed.

# Kubewarden Incubation Application
v1.6
This template provides the project with a framework to inform the TOC of their conformance to the Incubation Level Criteria. 

Project Repo(s): https://github.com/kubewarden/community
Project Site:  https://kubewarden.io/
Sub-Projects: https://github.com/kubewarden/community?tab=readme-ov-file#repositories
Related Projects: N/A  _(List any projects you or your core maintainers operate in other foundations (Linux Foundation, Apache, Eclipse, etc.) that are technically related to this project. Include: project name, foundation, and relationship.)_
Communication:  [#kubewarden](https://kubernetes.slack.com/archives/C01T3GTC3L7) on Kubernetes Slack, [#kubewarden-dev](https://kubernetes.slack.com/archives/C03L52JRAFM) on Kubernetes Slack, cncf-kubewarden-maintainers as mail username at lists.cncf.io.

Project points of contacts: https://github.com/orgs/kubewarden/teams/maintainers or cncf-kubewarden-maintainers as mail username at lists.cncf.io.

- [ ] (Post Incubation only) [Book a meeting with CNCF staff](http://project-meetings.cncf.io) to understand project benefits and event resources. 

## Incubation Criteria Summary for Kubewarden

### Application Level Assertion

- [x] This project is currently Sandbox, accepted on 2022-06-14, and applying to Incubation.
- [x] This project is applying to join the CNCF at the Incubation level.

### Adoption Assertion

_The project has been adopted by the following organizations in a testing and integration or production capacity:_
See [ADOPTERS.md](https://github.com/kubewarden/community/blob/main/ADOPTERS.md)

**Adopter Interviews:** Submit 5-7 adopters via the [Adopter Interview Questionnaire](https://docs.google.com/forms/d/1n1oLC6IKj5-7S_xeEjIdEjbtS9SWniuAo7IIOyLFuK8/edit) prior to DD assignment. See the [definition of an adopter](https://github.com/cncf/toc/blob/main/FAQ.md#what-is-the-definition-of-an-adopter) for guidance on who qualifies.

## Application Process Principles

### Suggested

- [x] **Engage with domain-specific TAG(s) to present the technical architecture of the project.**
  - This occurred on DD-MMM-YYYY and can be discovered at $LINK.

This was completed and occurred on 09-Jun-2022, and can be discovered at https://github.com/cncf/tag-security/issues/899.
slide deck: https://drive.google.com/file/d/1JeC0GsDHP0X5dnTjhZTAL_Ruc1y12PJt/view?usp=sharing
recording: https://www.youtube.com/watch?v=K-8Nepk7Zyc
General Technical Review: https://github.com/kubewarden/community/blob/main/general-technical-questions.md


### Required


- [x] **Complete a [General Technical Review (GTR)](../toc_subprojects/project-reviews-subproject/general-technical-questions.md).**
  - This was most recently revised on DD-MMM-YYYY and can be discovered at $LINK.

Revised on 2025-05-15. The link to the document: https://github.com/kubewarden/community/blob/main/general-technical-questions.md

- [ ] **Complete a [Governance Review](../toc_subprojects/project-reviews-subproject/governance-review-template.md).**
  - This was most recently revised on DD-MMM-YYYY and can be discovered at $LINK.



- [x] **All project metadata and resources are [vendor-neutral](https://contribute.cncf.io/maintainers/community/vendor-neutrality/).**

https://github.com/kubewarden/community/blob/main/GOVERNANCE.md

- [x] **Review and acknowledgement of expectations for [Sandbox](https://sandbox.cncf.io) projects and requirements for moving forward through the CNCF Maturity levels.**		
- Met during Project's application on 14-Jul-2022

[cncf/sandbox#213](https://github.com/cncf/sandbox/issues/213)

[Results from Sandbox Inclusion Meeting, 6/14](https://lists.cncf.io/g/cncf-toc/topic/results_from_sandbox/91754161?p=,,,20,0,0,0::recentpostdate/sticky,,,20,2,0,91754161,previd%3D1655255497954310253,nextid%3D1653320044101394801&previd=1655255497954310253&nextid=1653320044101394801)


- [ ] **Due Diligence Review.**

Completion of this due diligence document, resolution of concerns raised, and presented for public comment satisfies the Due Diligence Review criteria.

- [x] **Additional documentation as appropriate for project type, e.g.: installation documentation, end user documentation, reference implementation and/or code samples.**

https://docs.kubewarden.io/
https://github.com/kubewarden/community

## Governance and Maintainers

Note: this section may be augmented by the completion of a Governance Review from the Project Reviews subproject if completed as a suggested item prior to application.

### Suggested

- [ ] **Complete a Governance Review with the Project Reviews subproject**

 

- [x] **Governance has continuously been iterated upon by the project as a result of their experience applying it, with the governance history demonstrating evolution of maturity alongside the project's maturity evolution.**

 

- [x] **Clear and discoverable project governance documentation.**

https://github.com/kubewarden/community/blob/main/GOVERNANCE.md

- [x] **Governance is up to date with actual project activities, including any meetings, elections, leadership, or approval processes.**

 

- [x] **Governance clearly documents [vendor-neutrality](https://contribute.cncf.io/maintainers/community/vendor-neutrality/) of project direction.**

https://docs.kubewarden.io/admission-controller/1.36/en/introduction.html#_vendor_neutrality
https://github.com/kubewarden/community/blob/befc1760d5c93e9e497eb9f029b01443ce79951f/GOVERNANCE.md


- [x] **Document how the project makes decisions on leadership, contribution acceptance, requests to the CNCF, and changes to governance or project goals.**

https://github.com/kubewarden/community/blob/main/GOVERNANCE.md

- [ ] **Document how role, function-based members, or sub-teams are assigned, onboarded, and removed for specific teams (example: Security Response Committee).**

See [CONTRIBUTOR_LADDER.md](https://github.com/kubewarden/community/blob/main/CONTRIBUTOR_LADDER.md). The project doesn't have many subteams right now, but as an example, see "Mentees".

- [x] **Document a complete maintainer lifecycle process (including roles, onboarding, offboarding, and emeritus status).**

In the Kubewarden team, we have only two roles that are applied to the whole project. These role definitions are defined in the [GOVERNANCE.md](https://github.com/kubewarden/community/blob/main/GOVERNANCE.md) file

- [x] **Demonstrate usage of the maintainer lifecycle with outcomes, either through the addition or replacement of maintainers as project events have required.**

https://github.com/kubewarden/community/blob/main/GOVERNANCE.md

- [x] **If the project has subprojects: subproject leadership, contribution, maturity status documented, including add/remove process.**

 

### Required

- [x] **Document complete list of current maintainers, including names, contact information, domain of responsibility, and affiliation.**

https://github.com/kubewarden/community/blob/main/MAINTAINERS.md

- [x] **A number of active maintainers which is appropriate to the size and scope of the project.**

https://github.com/kubewarden/community/blob/main/MAINTAINERS.md

- [x] **Code and Doc ownership in Github and elsewhere matches documented governance roles.**

We use the CODEOWNERS file in all of the repositories. Some examples:
https://github.com/kubewarden/kubewarden-controller/blob/main/CODEOWNERS
https://github.com/kubewarden/docs/blob/main/CODEOWNERS

- [x] **Document adoption and adherence to the CNCF Code of Conduct or the project's CoC which is based off the CNCF CoC and not in conflict with it.**

https://github.com/kubewarden/community/blob/main/CODE_OF_CONDUCT.md

- [x] **CNCF Code of Conduct is cross-linked from other governance documents.**

https://github.com/kubewarden/community/blob/main/CODE_OF_CONDUCT.md

- [x] **All subprojects, if any, are listed.**

https://github.com/kubewarden/community?tab=readme-ov-file#repositories

## Contributors and Community

Note: this section may be augmented by the completion of a Governance Review from the Project Reviews subproject.

### Suggested

- [x] **Contributor ladder with multiple roles for contributors.**

https://github.com/kubewarden/community/blob/main/CONTRIBUTOR_LADDER.md

### Required

- [x] **Clearly defined and discoverable process to submit issues or changes.**

We use GitHub to submit issues and changes via PRs. Contributors can get in contact with us at [#kubewarden-dev](https://kubernetes.slack.com/archives/C03L52JRAFM) on the Kubernetes Slack server, as well as cncf-kubewarden-maintainers@lists.cncf.io. We also have the [CONTRIBUTING.md](https://github.com/kubewarden/community/blob/main/CONTRIBUTING.md) file

- [x] **Project must have, and document, at least one public communications channel for users and/or contributors.**

We use [#kubewarden-dev](https://kubernetes.slack.com/archives/C03L52JRAFM), [#kubewarden](https://kubernetes.slack.com/archives/C01T3GTC3L7) on the Kubernetes Slack server, as well as [cncf-kubewarden-maintainers@lists.cncf.io](mailto:cncf-kubewarden-maintainers@lists.cncf.io). These are linked in [https://kubewarden.io/](https://kubewarden.io/). We also describe how to contribute to the project in the [CONTRIBUTING.md](https://github.com/kubewarden/community/blob/main/CONTRIBUTING.md#feedback) file

- [x] **List and document all project communication channels, including subprojects (mail list/slack/etc.).  List any non-public communications channels and what their special purpose is.**

https://github.com/kubewarden/community/tree/main?tab=readme-ov-file#community

- [x] **Up-to-date public meeting schedulers and/or integration with CNCF calendar.**

We have our community calls listed in the [CNCF calendar](https://www.cncf.io/calendar/) and in our [community repository](https://github.com/kubewarden/community/tree/main?tab=readme-ov-file#community-meeting)

- [x] **Documentation of how to contribute, with increasing detail as the project matures.**

https://github.com/kubewarden/community/blob/main/CONTRIBUTING.md
https://github.com/kubewarden/adm-controller/blob/main/CONTRIBUTING.md
https://github.com/kubewarden/sbomscanner/blob/main/CONTRIBUTING.md
https://github.com/kubewarden/docs/blob/main/CONTRIBUTING.adoc

- [x] **Demonstrate contributor activity and recruitment.**

https://kubewarden.devstats.cncf.io/
https://kubewarden.devstats.cncf.io/d/74/contributions-chart?orgId=1

## Engineering Principles

### Suggested

- [x] **Roadmap change process is documented.**

https://github.com/kubewarden/community/blob/main/ROADMAP.md

- [x] **History of regular, quality releases.**

We follow a monthly release cadence. The releases, including the release candidates, can be found in the GitHub repositories.
We also write a blog post for each release.
https://github.com/kubewarden/adm-controller/releases
https://github.com/kubewarden/sbomscanner/releases

Note that older releases could be found in now archived (now incorporated in the `adm-controller`  ) repositories : 

https://github.com/kubewarden/policy-server/releases
https://github.com/kubewarden/kwctl/releases
https://github.com/kubewarden/audit-scanner/releases

### Required 

- [x] **Document project goals and objectives that illustrate the project’s differentiation in the Cloud Native landscape as well as outlines how this project fulfills an outstanding need and/or solves a problem differently. _This can also be satisfied by completing a General Technical Review._**
  - _If applicable_ a General Technical Review was completed/updated on DD-MMM-YYYY, and can be discovered at $LINK.

The link to the GTR document: https://github.com/kubewarden/community/blob/main/general-technical-questions.md. Revised on 2025-05-15. 

- [x] **Document what the project does, and why it does it - including viable cloud native use cases. This can also be satisfied by completing a General Technical Review.**

General Technical Review: [https://github.com/kubewarden/community/blob/main/general-technical-questions.md](https://github.com/kubewarden/community/blob/main/general-technical-questions.md)

- [x] **Document and maintain a public roadmap or other forward looking planning document or tracking mechanism.**

We use Github milestones to define the roadmap for future releases. This can be found in the project board at the "Milestone" tab
[Kubewarden Admission Controller (view)](https://github.com/orgs/kubewarden/projects/6/views/12)
The roadmap strategy is documented here: https://github.com/kubewarden/community/blob/main/ROADMAP.md

- [x] **Document overview of project architecture and software design that demonstrates viable cloud native use cases, as part of the project's documentation. _This can also be satisfied by completing a General Technical Review and capturing the output in the project's documentation._**
  - _If applicable_ a General Technical Review was completed/updated on DD-MMM-YYYY, and can be discovered at $LINK.

The link to the GTR document: https://github.com/kubewarden/community/blob/main/general-technical-questions.md. Revised on 2025-05-15. 
https://docs.kubewarden.io/explanations/architecture

- [x] **Document the project's release process.**

https://github.com/kubewarden/adm-controller/blob/main/CONTRIBUTING.md#releasing
https://github.com/kubewarden/helm-charts/blob/main/CONTRIBUTING.md

## Security

Note: this section may be augmented by a joint-assessment performed by TAG Security and Compliance if completed as a suggested item prior to application.

### Suggested

- [ ] **Complete a joint security assessment with TAG Security and Compliance**

 

### Required

- [x] **Clearly defined and discoverable process to report security issues.**

https://github.com/kubewarden/community/blob/main/SECURITY.md
https://docs.kubewarden.io/disclosure

- [x] **Enforcing Access Control Rules to secure the code base against attacks (Example: two factor authentication enforcement, and/or use of ACL tools.)**

As a Github hosted project, we rely on the Github authentication mechanisms. Most of the maintainers use two factor authentication and sign commits and tags with GPG keys

- [x] **Document assignment of security response roles and how reports are handled.** _(This is distinct from the security reporting process above — document who is responsible for triaging and responding to reports, not just where to send them. For reference, see the [Kubernetes Security Response Committee](https://github.com/kubernetes/committee-security-response) as a model for named membership, documented responsibilities, and a clear escalation path.)_

This is described in the [SECURITY.md](https://github.com/kubewarden/community/blob/main/SECURITY.md) file. The security is issue are handle by the maintainers of the project via the communication channel described in the documentation

- [x] **Document [Security Self-Assessment](https://tag-security.cncf.io/community/assessments/guide/self-assessment/).**

https://github.com/kubewarden/community/blob/main/security-self-assesment.md

- [x] **Achieve the [Open Source Security Foundation (OpenSSF) Best Practices passing badge](https://www.bestpractices.dev/).** _(Link to your bestpractices.dev project page below.)_

https://www.bestpractices.dev/en/users/23101

## Ecosystem

### Suggested

N/A

### Required

- [x] **Publicly documented list of adopters, which may indicate their adoption level (dev/trialing, prod, etc.)**

https://github.com/kubewarden/community/blob/main/ADOPTERS.md

- [x] **Used in appropriate capacity by at least 3 independent + indirect/direct adopters, (these are not required to be in the publicly documented list of adopters)**

The project provided the TOC with a list of adopters for verification of use of the project at the level expected, i.e. production use for graduation, dev/test for incubation: https://github.com/kubewarden/community/blob/main/ADOPTERS.md

- [ ] **TOC verification of adopters.**

Refer to the Adoption portion of this document


- [x] **Clearly documented integrations and/or compatibility with other CNCF projects as well as non-CNCF projects.**

Docs page listing the dependencies used: https://docs.kubewarden.io/reference/dependency-matrix

## Additional Information



</details>


---

In summary, I still think the most important for us now is to get the number of the adopters willing to be interviewed by CNCF. The other minor tasks depends on us and they are not that hard to do. Most of they are documentation improvements and doing the missing review. And we should evaluate if the feedback the k3s got from the DD review apply for us as well.  

---

**Comment by viccuad** [comment]
Thanks @jvanz for the review, and thanks for providing the incubation application draft filled. It was previously in the issue description, but got edited away. Since it is versioned, I think it's a good idea to keep it around.

I've opened sub-issues for things you list that are actionable to us:

- https://github.com/kubewarden/community/issues/94
- https://github.com/kubewarden/community/issues/95
- ~~Document how role, function-based members, or sub-teams are assigned, onboarded, and removed for specific teams (example: Security Response Committee).~~ Done, updated submission draft.
- https://github.com/kubewarden/community/issues/96
- (optional) Complete a join security assessment with TAG Security and Compliance
- (nice to have) Refresh Security self-assessment with SBOM Scanner
- (nice to have) Refresh CLOMonitor with SBOM Scanner
- (nice to have) Refresh [OpenSSF best practices](https://www.bestpractices.dev/en/users/23101) with SBOM Scanner
